changelog.d/10115.bugfix

```diff
@@ -0,0 +1,2 @@
+Fix a bug that prevented the 'ip_range_whitelist' configuration to work for federation and identity servers.
```

Comment:
```suggestion
Fix a bug introduced in Synapse v1.25.0 that prevented the `ip_range_whitelist` configuration option from working for federation and identity servers. Contributed by @mikure.
```

Reply:
This feels a bit odd to me since this wasn't a thing that ever worked -- I don't think it is really a bug, but a new feature? I guess it depends on how you look at it.

Reply:
I had the same thought, but the sample config says of `ip_range_whitelist`:

> # List of IP address CIDR ranges that should be allowed for federation,
> # identity servers, push servers, ...

which is just wrong. So whether it be in the docs or the impl, it seems there was a bug.

Reply:
Ah, fair enough! I didn't realize the docs were that specific. 👍

Reply:
Thank you for your feedback.
For me it felt like a bug, as the `ip_range_whitelist` did not work for my test setup, although the comment claimed otherwise.